File: src/worker/core/draft/logLotteryChances.java
// This code is translated from TypeScript to Java.
import logAction from "./logAction";
import logLotteryTxt from "./logLotteryTxt";

import java.util.List;

public class LotteryLogger {

    public static void logLotteryChances(
            double[] chances,
            List<TeamFiltered> teams,
            List<List<DraftPickWithoutKey>> draftPicksIndexed,
            Conditions conditions) {
        
        for (int index = 0; index < chances.length; index++) {
            if (index < teams.size()) {
                int originalTeamId = teams.get(index).getTid();
                DraftPickWithoutKey draftPick = draftPicksIndexed.get(originalTeamId).get(1);

                if (draftPick != null) {
                    int teamId = draftPick.getTid();
                    String text = logLotteryTxt(teamId, "chance", chances[index]);
                    logAction(teamId, text, 0, conditions);
                }
            }
        }
    }
}

File: src/worker/core/draft/logLotteryChances.clj
;; Clojure code translation for TypeScript source code
(ns lottery.core
  (:require [lottery.log-action :refer [log-action]]
            [lottery.log-lottery-txt :refer [log-lottery-txt]]))

(defn log-lottery-chances
  [chances teams draft-picks-indexed & [conditions]]
  (doseq [index (range (count chances))]
    (when (< index (count teams))
      (let [original-tid (:tid (nth teams index))
            draft-pick (get-in draft-picks-indexed [original-tid 1])]
        (when draft-pick
          (let [tid (:tid draft-pick)
                txt (log-lottery-txt tid "chance" (nth chances index))]
            (log-action tid txt 0 conditions)))))))
